cpp/src/parquet/column_writer_test.cc

```diff
@@ -254,7 +254,8 @@ class TestPrimitiveWriter : public PrimitiveTypedTest<TestType> {
     for (size_t i = 0; i < this->values_.size(); i++) {
       if (comparator->Compare(this->values_[i], this->values_out_[i]) ||
           comparator->Compare(this->values_out_[i], this->values_[i])) {
-        std::cout << "Failed at " << i << std::endl;
+        // std::cout << "Failed at " << i << std::endl;
+        ARROW_SCOPED_TRACE("Failed at ", i);
```

Comment:
The message should be more like `ARROW_SCOPED_TRACE("i=",i);`.  The trace should tell use the parameter values, it doesn't need to include "failed at" as that will be implied (the trace info is only shown on a failure).